src/linprog.jl

```diff
@@ -1,16 +1,67 @@
 export linprog
 
-const MatrixOrOperator = Union{AbstractMatrix,AbstractLinearOperator}
+const MatrixOrOperator = Union{AbstractMatrix, AbstractLinearOperator}
 
 """
-    linprog(c, A, b)
+    linprog(c; x0, A, blow, bup, Aeq, beq, lvar, uvar)
+
+Minimize cᵀ x subject to any combination of the optional constraints
+blow ≤ A x ≤ bup, Aeq x = beq and lvar ≤ x ≤ uvar.
+Initial solution is optional and set to zeros by default.
+One sided inequalities like A x ≤ bup are equivalent to [-∞,…,-∞] ≤ A x ≤ bup.
 
-Minimize the linear problem cᵀ x with constraints A x ≤ b.
 """
-function linprog(c::AbstractVector, A::MatrixOrOperator, b::AbstractVector)
+function linprog(c::AbstractVector; x0::AbstractVector = zeros(length(c)), A::MatrixOrOperator = zeros(0,0),
```

Comment:
A few things:
- Change `up` to `upp`;
- Change initial values: `A = zeros(0, length(c))`, `Aeq = zeros(0, length(c))`;
- Add `@assert` checks for size consistency;
- `ADNLPModel` defines the number of constraints according to the size of `lcon` and `ucon`, so only line 33 is necessary.

Reply:
Also, change `A` to `Aineq`.